types/dockerode/index.d.ts

```diff
@@ -1353,7 +1353,7 @@ declare class Dockerode {
     getEvents(options?: Dockerode.GetEventsOptions): Promise<NodeJS.ReadableStream>;
 
     pull(repoTag: string, options: {}, callback: Callback<any>, auth?: {}): Dockerode.Image;
-    pull(repoTag: string, options: {}, auth?: {}): Promise<any>;
+    pull(repoTag: string, options?: {}): Promise<any>;
```

Comment:
Should this be an additional signature so we have both?

```typescript
pull(repoTag: string, options: {}, auth?: {}): Promise<any>;
pull(repoTag: string, options?: {}): Promise<any>;
```

Reply:
I believe that the previous signature doesn't work. The source code of the function is the following:

```javascript
Docker.prototype.pull = function(repoTag, opts, callback, auth) {
  var args = util.processArgs(opts, callback);

  var imageSrc = util.parseRepositoryTag(repoTag);
  args.opts.fromImage = imageSrc.repository;
  args.opts.tag = imageSrc.tag || 'latest';

  var argsf = [args.opts, args.callback];
  if (auth) {
    argsf = [auth, args.opts, args.callback];
  }
  return this.createImage.apply(this, argsf);
};
```

When callback is null or undefined, then the function returns a Promise.

By looking at this code, I think that if you try to use this signature `pull(repoTag: string, options: {}, auth?: {}): Promise<any>;` then the `auth` object will be assigned to the `callback` function and it won't work as expected.